src/app/app.service.ts

```diff
@@ -18,20 +18,30 @@ export class AppConfig {
   private _config
   private isBrowser
 
+  private PORTS_TO_EXCLUDE = ["80", "4000", "8080"];
+
   constructor(@Inject(PLATFORM_ID) private platformId, private injector: Injector) {
     this.isBrowser = isPlatformBrowser(platformId);
+
     // Identify hostname from request or client side
+    this.isBrowser = false;
     if (this.isBrowser) {
-      this.clientHostname = window.location.hostname
-    } else{
+      this.clientHostname = this.getHostName(window.location.hostname, window.location.port);
+    } else {
       let req = this.injector.get('request');
-      
-      this.clientHostname = req ? req.get('host') : '';
+      if(req) {
+        const requestHost = req.get("host").split(":");
+        this.clientHostname = (requestHost.length === 1) ?
+                                requestHost[0] :
+                                this.getHostName(requestHost[0], requestHost[1]);
+      }
+      else {
+        this.clientHostname = "";
+      }
+
```

Comment:
I noticed that we have two conditions here to determine the host name, and it depends on if `isBrowser` is true or false? Not sure I am understand the use cases for both. Any idea what the difference is?

I guess I am mostly asking because it appears we only did the filtering of the ports if `isBrowser=false`. And honestly not too sure what that was designed for in the first place.

Reply:
It has to do with server-side rendering, and when `isPlatformBrowser` is false, the window variable is undefined, so I imagine that is the reason for the if/then statement.